scripts/xpcs_online_boost_client_confclient.py

```diff
@@ -0,0 +1,160 @@
+                    'atype': atype,
+                    "qmap": qmap_file,
+                    "raw": raw_file,
+                    "output": output_dir,
+                    "batch_size": 8,
+                    "gpu_id": args.gpu_flag,
+                    "verbose": args.verbose,
+                    "masked_ratio_threshold": 0.75,
+                    "use_loader": True,
+                    "begin_frame": 1,
+                    "end_frame": -1,
+                    "avg_frame": 1,
+                    "stride_frame": 1,
+                    "overwrite": True,
+            },
+
+            'pilot': {
+                # This is the directory which will be published
+                'dataset': dataset_dir,
+                # Old index, switch back to this when we want to publish to the main index
+                'index': '6871e83e-866b-41bc-8430-e3cf83b43bdc',
+                # Test Index, use this for testing
+                # 'index': '2e72452f-e932-4da0-b43c-1c722716896e',
+                'project': 'xpcs-8id',
+                'source_globus_endpoint': depl_input['input']['globus_endpoint_proc'],
+                'source_collection_basepath': str(deployment.staging_collection.path),
+                # Extra groups can be specified here. The XPCS Admins group will always
+                # be provided automatically.
+                'groups': [args.group] if args.group else [],
+            },
+
+            'source_transfer': {
+                'source_endpoint_id': deployment.source_collection.uuid,
+                'destination_endpoint_id': deployment.staging_collection.uuid,
+                'transfer_items': [
+                    {
+                        'source_path': args.raw,
+                        'destination_path': deployment.staging_collection.to_globus(raw_file),
+                    },
+                    {
+                        'source_path': args.hdf,
+                        'destination_path': deployment.staging_collection.to_globus(input_hdf_file),
+                    },
+                    {
+                        'source_path': args.qmap,
+                        'destination_path': deployment.staging_collection.to_globus(qmap_file),
+                    }
+                ],
+            },
+
+            'proc_dir': dataset_dir,
+            'metadata_file': input_hdf_file,
+            'hdf_file': output_hdf_file,
+            'execution_metadata_file': execution_metadata_file,
+
+            # funcX endpoints
+            'funcx_endpoint_non_compute': depl_input['input']['funcx_endpoint_non_compute'],
+            'funcx_endpoint_compute': depl_input['input']['funcx_endpoint_compute'],
+        }
+    }
+
+    client = XPCSBoost(login_manager=callback_login_manager)
+
+    corr_run_label = pathlib.Path(hdf_name).name[:62]
+    flow_run = client.run_flow(flow_input=flow_input, label=corr_run_label, tags=['aps', 'xpcs'])
+
+    print('run_id : ' + flow_run['action_id'])
```

Comment:
Could we re-use the original xpcs-online-boost client instead of copying the code into a new module? Ideally something in the deployment would mark it as a service account and make it expect the CLIENT_ID and CLIENT_SECRET from the env.

Reply:
Good suggestion! These changes were minimal after cleaning it up.